internal/proxysql/core.go

```diff
@@ -1,150 +1,231 @@
 		if err != nil {
-			slog.Error("Commands failed", slog.String("commands", command), slog.Any("error", err))
+			slog.Error("Error in addPod()", slog.Any("err", err))
 		}
 	}
 
-	// Write the new checksum to the file for the next run
-	err = os.WriteFile(checksumFile, []byte(digest), 0o600)
-	if err != nil {
-		slog.Error("Failed to write to checksum file", slog.String("file", checksumFile), slog.Any("error", err))
+	// Pod is shutting down. Only run this for core pods, as satellites don't need special considerations when
+	// they leave the cluster.
+	if oldpod.Status.Phase == "Running" && newpod.Status.Phase == "Failed" {
+		err := p.removePodFromCluster(oldpod)
+		if err != nil {
+			slog.Error("Error in removePod()", slog.Any("err", err))
+		}
 	}
-
-	slog.Info("Commands ran", slog.String("commands", strings.Join(commands, "; ")))
 }
 
-func GetCorePods(settings *configuration.Config) ([]PodInfo, error) {
-	app := settings.Core.PodSelector.App
-	component := settings.Core.PodSelector.Component
-	namespace := settings.Core.PodSelector.Namespace
-
-	config, err := rest.InClusterConfig()
-	if err != nil {
-		return nil, err
-	}
+// Add the new pod to the cluster.
+//   - If it's a core pod, add it to the proxysql_servers table
+//   - if it's a satellite pod, run the commands to accept it to the cluster
+func (p *ProxySQL) addPodToCluster(pod *v1.Pod) error {
+	slog.Info("Pod joined the cluster", slog.String("name", pod.Name), slog.String("ip", pod.Status.PodIP))
 
-	clientset, err := kubernetes.NewForConfig(config)
-	if err != nil {
-		return nil, err
-	}
+	commands := []string{}
 
-	pods, _ := clientset.CoreV1().Pods(namespace).List(context.TODO(), metav1.ListOptions{
-		LabelSelector: fmt.Sprintf("app=%s,component=%s", app, component),
-	})
+	// If the new pod is a core pod, delete the default entries in the proxysql_server list and add the new pod to it.
+	if pod.Labels["component"] == "core" {
+		// TODO: maybe make this configurable, not everyone will name the service this.
+		commands = append(commands, "DELETE FROM proxysql_servers WHERE hostname = 'proxysql-core'")
 
-	var corePods []PodInfo
-	for _, pod := range pods.Items {
-		corePods = append(corePods, PodInfo{PodIP: pod.Status.PodIP, Hostname: pod.Name, UID: string(pod.GetUID())})
+		cmd := fmt.Sprintf("INSERT INTO proxysql_servers VALUES ('%s', 6032, 0, '%s')", pod.Status.PodIP, pod.Name)
+		commands = append(commands, cmd)
 	}
 
-	return corePods, err
-}
-
-func calculateChecksum(pods []PodInfo) string {
-	data := []string{}
+	commands = append(commands,
+		"LOAD PROXYSQL SERVERS TO RUNTIME",
+		"LOAD ADMIN VARIABLES TO RUNTIME",
+		"LOAD MYSQL VARIABLES TO RUNTIME",
+		"LOAD MYSQL SERVERS TO RUNTIME",
+		"LOAD MYSQL USERS TO RUNTIME",
+		"LOAD MYSQL QUERY RULES TO RUNTIME",
+	)
 
-	for _, pod := range pods {
-		data = append(data, fmt.Sprintf("%s:%s:%s", pod.PodIP, pod.Hostname, pod.UID))
+	for _, command := range commands {
+		_, err := p.conn.Exec(command)
```

Comment:
And then here:

```suggestion
		_, err := p.conn.Exec(command[0].(string), command[1:]...)
```

Reply:
i'm not sure what this line means, can you explain real quick?